docs/object_code.md

```diff
@@ -0,0 +1,272 @@
+# Exo Object Code Syntax
+
+In Exo, object code can be defined using Python-like syntax with specific annotations and constructs to model low-level programming concepts.
+
+This documentation explains Exo's object code syntax using the following example of a 1D convolution operation:
+
+```python
+@proc
+def generic_conv1d(
+    data: i32[IC, N] @ DRAM,
+    kernels: i32[OC, IC, W] @ DRAM,
+    out: i32[OC, N] @ DRAM,
+):
+    # Perform the convolution
+    for i in seq(0, OC):
+        for j in seq(0, N):
+            # Zero out the output memory
+            out[i, j] = 0.0
+            for c in seq(0, IC):
+                for r in seq(0, W):
+                    y: i32
+                    if j + r < N:
+                        y = data[c, j + r]
+                    else:
+                        y = 0
+                    out[i, j] += kernels[i, c, r] * y
+```
+
+## Table of Contents
+
+- [Annotations and Decorators](#annotations-and-decorators)
+  - [`@proc` Decorator](#proc-decorator)
+  - [Type and Memory Annotations](#type-and-memory-annotations)
+- [Procedure Arguments](#procedure-arguments)
+- [Variable Declarations](#variable-declarations)
+- [Memory Spaces](#memory-spaces)
+- [Loops](#loops)
+  - [`for` Loop Syntax](#for-loop-syntax)
+- [Conditional Statements](#conditional-statements)
+- [Operations and Assignments](#operations-and-assignments)
+- [Understanding the Example](#understanding-the-example)
+- [Conclusion](#conclusion)
+
+## Annotations and Decorators
+
+### `@proc` Decorator
+
+The `@proc` decorator is used to define an Exo procedure (analogous to a function in other programming languages). It indicates that the following function definition should be treated as Exo object code (not Python), which can be further optimized and transformed.
+
+```python
+@proc
+def function_name(arguments):
+    # Function body
+```
+
+### Type and Memory Annotations
+
+In Exo, types and memory spaces are explicitly annotated. The syntax is:
+
+```python
+name: type[size] @ memory
+```
+
+- **`name`**: The variable name.
+- **`type`**: The data type (e.g., `i32`, `f32`).
```

Comment:
Give a list of valid types (or tell where in the code to find the list of valid types). Their names may not be predictable to C programmers.